### PR TITLE
gui/user management: Big Refactor for more close to PS4. 

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- gui/user management: Big Refactor for more close to PS4.
reworks size and position of avatar.
Add box for avatar and user name.
reworks user name position.
limit input to 16 letter.
move menu from string to enum with switch.

# Result
image
image

### CODE REVIEW

File: vita3k/gui/src/user_management.cpp [hunk 1]
```diff
@@ -71,19 +77,15 @@ static bool init_avatar(GuiState &gui, EmuEnvState &emuenv, const std::string &u
...
-
-    // Resize for avatar
-    const auto ratio = std::min(avatar_size / static_cast<float>(width), avatar_size / static_cast<float>(height));
-    user_avatar.size = ImVec2(width * ratio, height * ratio);
-
-    // Calculate position for both avatar size
-    user_avatar.pos = ImVec2((avatar_size / 2.f) - (user_avatar.size.x / 2.f), (avatar_size / 2.f) - (user_avatar.size.y / 2.f));
-    user_avatar.small_pos = ImVec2((avatar_small_size / 2.f) - (user_avatar.small_size.x / 2.f), (avatar_small_size / 2.f) - (user_avatar.small_size.y / 2.f));
+    // Resize for all size of avatar
+    std::vector<AvatarSize> sizes = { SMALL, MEDIUM, BIG };
```
Comment (bookmist) [inline vita3k/gui/src/user_management.cpp:81]: std::array can be used here

Comment (Zangetsu38) [reply vita3k/gui/src/user_management.cpp:81]: no sur understund ? how can be used ?

Comment (Zangetsu38) [reply vita3k/gui/src/user_management.cpp]: Done

File: vita3k/gui/src/user_management.cpp [hunk 2]
```diff
@@ -232,91 +238,119 @@ static auto get_users_index(GuiState &gui, const std::string &user_name) {
...
     ImGui::BeginChild("##user_child", SIZE_USER, false, ImGuiWindowFlags_NoBackground | ImGuiWindowFlags_HorizontalScrollbar | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoSavedSettings);
 
-    const auto AVATAR_POS = ImVec2((SIZE_USER.x / 2) - (AVATAR_SIZE.x / 2.f), (((menu == "create" || menu == "edit") ? 48.f * SCALE.y : (SIZE_USER.y / 2) - (AVATAR_SIZE.y / 2.f))));
-    const auto NEW_USER_POS = AVATAR_POS.x - (gui.users.size() ? SPACE_AVATAR : 0.f);
-    const auto DELETE_USER_POS = AVATAR_POS.x + (SPACE_AVATAR * (gui.users.size()));
+    // Clear temp user
+    const auto clear_temp_user = [&]() {
+        temp = {};
+        gui.users_avatar["temp"] = {};
+        gui.users_avatar.erase("temp");
```
Comment (bookmist) [inline vita3k/gui/src/user_management.cpp:302]: No need to clear before erase

**Review by bookmist — APPROVED** [review]

File: vita3k/gui/src/user_management.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/gui/src/user_management.cpp:302]: is obligatory, in moment it run ths, picture is keep display
need clear texcture before can erase it

**Pull request merged**